src/native_module/utils.cljs

```diff
@@ -0,0 +1,22 @@
+(ns native-module.utils
+  (:require
+    [clojure.string :as string]
+    [promesa.core :as promesa]
+    [utils.transforms :as types]))
+
+(defn- promisify-callback
+  [res rej]
+  (fn [result]
+    (let [error (let [{:keys [error]} (types/json->clj result)]
+                  (when-not (string/blank? error)
+                    error))]
+      (if error (rej error) (res result)))))
+
+(defn promisify-native-module-call
+  [func & args]
+  (promesa/create
+   (fn [res rej]
+     (->> (promisify-callback res rej)
+          (conj [])
+          (concat args)
+          (apply func)))))
```

Comment:
@cammellos in order to use native module calls as promises and chain them, added this utility (used in `src/native_module/core.cljs`). Wanted to know if this would be a good way to handle this and whether there are better alternatives.

Reply:
it looks ok to me, maybe leave out the extra `conj` since it's redundant?

`(->> [(promisify....)]
         (concat args)
         ....)`